### ISSUE TITLE
loss function like has a bug

### ISSUE THREAD

**Issue by rrjia** [ultralytics/yolov5#1988]
## 🐛 Bug

utils/loss.py  line108
```python
gain[2:6] = torch.tensor(p[i].shape)[[3, 2, 3, 2]]  # xyxy gain
```
for yolov5s.yaml
if i=1
p[i].shape = 3,80,80,85
gain[2:6] = [85, 80, 85, 80]
```python
gain[2:6] = torch.tensor(p[i].shape)[[2, 1, 2, 1]]  # xyxy gain
```
it shoud be [80, 80, 80, 80]

**Comment by rrjia** [comment]
![image](https://user-images.githubusercontent.com/13146304/105154176-3a5cdc00-5b44-11eb-9115-8550c3e516f4.png)

**Comment by glenn-jocher** [comment]
@rrjia hi thanks for the bug report and the screenshot! We have updated the loss function recently, creating a new ComputeLoss() class to replace it. Can you verify that you see the same bug in the most recent code please?

You can update your code with `git pull`, or alternatively you can git clone `https://github.com/ultralytics/yolov5` again.

**Comment by glenn-jocher** [comment]
@rrjia if I try to reproduce by training with default settings and placing a breakpoint here, everything looks correct:

<img width="1700" alt="Screen Shot 2021-01-20 at 9 53 34 AM" src="https://user-images.githubusercontent.com/26833433/105214905-71f86380-5b05-11eb-81ed-877ce7310d1e.png">

**Comment by rrjia** [comment]
sorry it's my fault, the model get A list of length 3, i use y[0] input compute_loss function, thank you.
